File: two_sum.py
"""
Author: Abhiram Ravi Bharadwaj
Source: Leetcode
Problem: https://leetcode.com/problems/two-sum/
"""
def twoSum(self, nums, target: int):
	diff = {}
	for i in range(len(nums)):
		if nums[i] in diff:
			return [diff[nums[i]], i]
		diff[target - nums[i]] = i

def main():
	print(twoSum([2, 7, 11, 15], 9))

if __name__ == '__main__':
	main()

File: C++/twoSum.cpp
//  https://leetcode.com/problems/two-sum/
//  twoSum.cpp
//  C++
//
//  Created by Abhiram Bharadwaj on 5/4/20.
//  Copyright © 2020 Abhiram Bharadwaj. All rights reserved.
//

#include <stdio.h>
#include <vector>
#include <map>

using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    vector<int> result(2);
    map<int,int> sumMap;
    
    for (vector<int>::iterator it = nums.begin() ; it != nums.end(); ++it){
        if(sumMap.find(*it) != sumMap.end()){
            result[0] = sumMap[*it];
            result[1] = it - nums.begin();
            break;
        } else{
            sumMap[target - *it] = it - nums.begin();
        }
    }
    return result;
}
